R/facet-null.r

```diff
@@ -30,8 +30,10 @@ FacetNull <- ggproto("FacetNull", Facet,
     # Need the is.waive check for special case where no data, but aesthetics
     # are mapped to vectors
     if (is.waive(data) || empty(data))
-      return(cbind(data, PANEL = integer(0)))
-    data$PANEL <- 1L
+      return(cbind(data, PANEL = factor()))
```

Comment:
If I change it to this:

```R
    if (is.waive(data))
      return(data)
    if (empty(data))
      return(cbind(data, PANEL = factor()))
```

It errors on this plot:

```R
ggplot() + geom_point(aes(1:5, 1:5))
# Error in if (n == 0) { : argument is of length zero 
```

It appears that when it's a waiver object, still requires the `PANEL` to be cbinded to it.


```
> cbind(waiver(), PANEL=factor())
      PANEL
> str(cbind(waiver(), PANEL=factor()))
 list()
 - attr(*, "dim")= int [1:2] 0 2
 - attr(*, "dimnames")=List of 2
  ..$ : NULL
  ..$ : chr [1:2] "" "PANEL"
```

Should I just leave as is?